### PR TITLE
Bump GTK version requirement to 3.24

### PR DESCRIPTION
**Pull request by techee**
This is needed for https://github.com/geany/geany/pull/3316. I tried to remove all related GTK_CHECK_VERSION() except the one that #3316 eliminates to avoid merge conflicts.

### CODE REVIEW

**Comment by elextr** [comment]
LGBI, love removing olde version workarounds :-)

**Comment by techee** [comment]
I've just re-pushed: I removed one more use of GTK_CHECK_VERSION from this patch which is already removed by #3316 to avoid merge conflicts.

File: src/dialogs.c [hunk 1]
```diff
@@ -841,8 +827,8 @@ on_font_dialog_response(GtkDialog *dialog, gint response, gpointer user_data)
 		{
 			gchar *fontname;
 
-			fontname = gtk_font_selection_dialog_get_font_name(
-				GTK_FONT_SELECTION_DIALOG(ui_widgets.open_fontsel));
+			fontname = gtk_font_chooser_get_font(
+				GTK_FONT_CHOOSER(ui_widgets.open_fontsel));
```
Comment (kugel-) [inline src/dialogs.c:831]: `GTK_FONT_CHOOSER_DIALOG()`

File: src/dialogs.c [hunk 2]
```diff
@@ -883,8 +869,8 @@ void dialogs_show_open_font(void)
 
 		gtk_window_set_transient_for(GTK_WINDOW(ui_widgets.open_fontsel), GTK_WINDOW(main_widgets.window));
 	}
-	gtk_font_selection_dialog_set_font_name(
-		GTK_FONT_SELECTION_DIALOG(ui_widgets.open_fontsel), interface_prefs.editor_font);
+	gtk_font_chooser_set_font(
+		GTK_FONT_CHOOSER(ui_widgets.open_fontsel), interface_prefs.editor_font);
```
Comment (kugel-) [inline src/dialogs.c:873]: `GTK_FONT_CHOOSER_DIALOG`

**Review by kugel- — CHANGES_REQUESTED** [review]

File: src/dialogs.c (hunk 1 above)
Comment (techee) [reply src/dialogs.c:831]: I think I did it right. Notice the original code - if first defines
```
#define GTK_FONT_SELECTION_DIALOG GTK_FONT_CHOOSER_DIALOG
```
but then, when it's called, there's an extra `GTK_FONT_CHOOSER()` which overrides it

Comment (kugel-) [reply src/dialogs.c:831]: Okay, I didnt see the calls inside the macros.

But still, they are constructed with `gtk_font_chooser_dialog_new()` so I would think `GTK_FONT_CHOOSER_DIALOG()` is correct anyway

**Comment by elextr** [comment]
This mixes two things:

- configuration requiring 3.24 which is absolutely required since currently it only requires 3.0 which is completely wrong.
- removal of various workarounds

The first is uncontroversial, a bugfix, and needed for release 2.0.  The second is nice to have, but not absolutely required for 2.0 AFAICT.

Perhaps it should be split.

**Comment by techee** [comment]
> Perhaps it should be split.

Can do if this is the preferred solution.

**Comment by eht16** [comment]
But why splitting? IMO it causes only extra work and we want to remove the workarounds anyway if they are not necessary any more.

Sure there is some risk of breakage but in this case this is probably tolerable especially since we will probably use and test Geany in the next days even more than usually.

**Comment by elextr** [comment]
Splitting is just to get the essential part done whilst there are still changes requested on the other part.  I didn't want to say override the changes requested.

**Comment by eht16** [comment]
@techee I just found another place to update, seems like we have forgotten it already the last time (:smile:): https://github.com/geany/geany/blob/master/doc/geany.txt?plain=1#L106

**Comment by techee** [comment]
> @techee I just found another place to update, seems like we have forgotten it already the last time (😄):

Good, thanks to this we'll have a smaller diff now bumping just the major version :-)

Which brought me to one more place - we can also drop some settings from `geany.css`!

Will do both.

**Comment by techee** [comment]
> Which brought me to one more place - we can also drop some settings from geany.css!

And also `geany-3.0.css` and `geany-3.20.css` can be dropped which I did in the last commit.

**Comment by eht16** [comment]
Awesome, thanks!

I'd vote for early merge to have as much testing via master as possible.

**Comment by elextr** [comment]
Works for me, happy for you to override the requested changes.

**Review by kugel- — APPROVED** [review]
Sorry

**Pull request merged**